apps/omg_watcher/lib/exit_processor.ex

```diff
@@ -294,6 +305,27 @@ defmodule OMG.Watcher.ExitProcessor do
     {:reply, canonicity_result, state}
   end
 
+  def handle_call({:create_challenge, Utxo.position(blknum, txindex, oindex) = exiting_utxo_pos}, _from, state) do
+    with spending_blknum_response <- exiting_utxo_pos |> Utxo.Position.to_db_key() |> OMG.DB.spent_blknum(),
```

Comment:
expecting something like this
```
resonse = with {:ok, data1} <- Outsite.getData(),
{:ok, data2} <- Outsite2.getdData(),
{:ok, data3} <- Outsite3.getData(), 
  do: Core.get_challenge(data1,data2,data3)
{:reply, response}
```

Reply:
I don't see now how to get there

Reply:
```
  with spending_blknum_response <- exiting_utxo_pos |> Utxo.Position.to_db_key() |> OMG.DB.spent_blknum(),
         %{txhash: txhash} <- OMG.Watcher.DB.Transaction.get_by_position(blknum, txindex),
         {:ok, exit_id} <- OMG.Eth.RootChain.get_standard_exit_id(txhash, oindex),
         {:ok, raw_spending_proof, exit_info} <- Core.get_challange_data(spending_blknum_response, exiting_utxo_pos, state) do
      spending_proof =
        case raw_spending_proof do
          blknum when is_number(blknum) ->
            {:ok, hashes} = OMG.DB.block_hashes([blknum])
            {:ok, [spending_block]} = OMG.DB.blocks(hashes)
            spending_block

          signed_tx ->
            signed_tx
        end

      {:reply, {:ok, Core.create_challenge(exit_info, spending_proof, exiting_utxo_pos, exit_id)}, state}
    end
```

Reply:
The run of this code(outside call to DB):
```
{:ok, hashes} = OMG.DB.block_hashes([blknum])
            {:ok, [spending_block]} = OMG.DB.blocks(hashes)
            spending_block
```
depends of `raw_spending_proof`.

Reply:
And that is a problem, get_challange_data returns two different types and shell have logic.
something like this:
```
{:ok, spending_blknum_response} <- exiting_utxo_pos |> Utxo.Position.to_db_key() |> OMG.DB.spent_blknum(),
{:ok, exit_id} <- OMG.Eth.RootChain.get_standard_exit_id(txhash, oindex),
{:ok, blknums} <- Core.get_challenges_blknums(exiting_utxo_pos, state),
{:ok, hashes} <- OMG.DB.block_hashes(blknums),
{:ok, spending_blocks} <- OMG.DB.blocks(hashes), 
   do: Core.create_challenge(spending_blocks, {exit_id, exiting_utxo_pos}, state)
```

Reply:
Yes, it's returning two types. Your propose is not resolving this problem. Your propose cares only about one type(blknum)

Reply:
not exactly, I propose that get_challenges_blknums can return empty hashes. create_challenge get potential spending blocks (can be empty), and do rest of the job.

Reply:
Still, I think your proposal is not going to work. You're missing staff about two types blknum and transaction

Reply:
as discussed, let's leave a todo/chore and move on